### ISSUE TITLE
ERROR: MethodError: no method matching iterate(::Nothing)

### ISSUE THREAD

**Issue by PuYuuu** [lassepe/peters2024ral-code#1]
Hi, when I execute _**demo_setup = ContingencyGames.Examples.setup_crosswalk_demo()**_, it prompts: _**ERROR: MethodError: no method matching iterate(::Nothing)**_ . And my OS version is WIN10, julia 1.10.0 . The complete error message is as follows:
![image](https://github.com/user-attachments/assets/be13182d-1041-4954-a021-135fbcd2de56)
Unfortunately I am not familiar with the julia language, could you give me some advice how to fix this error? Thank you

**Comment by lassepe** [comment]
Thank you for this question. I can see that the error message could be more informative. I hope this heps:

**TL;DR**: I suspect that you forgot to set the license key for PATH. Did you follow the instruction in the README?

**Detailed response**:

The error you are seeing comes from this line:
https://github.com/lassepe/peters2024ral-code/blob/e6213fa9c472099f5081523b480ece232ee2ebe9/src/Solver/mcp_solver.jl#L1015
It fails because `raw_solution.z` is nothing. The function that populates `raw_solution.z` uses PATHSolver.jl under the hood. That library returns `nothing` when no license was set or the license is invalid:

https://github.com/chkwon/PATHSolver.jl/blob/db02fc9f4285f1dec3f3db4ba4faed3acfd53552/src/C_API.jl#L746

If you follow the instructions in the README of how to set the license (there is a free license available for PATH), this issue should go away.

Let me know if you run into any other problems.

**Comment by lassepe** [comment]
Feel free to re-open if the issue persists.

**Comment by PuYuuu** [comment]
Thanks for your detailed reply. I did miss export PATH_LICENSE_STRING, and I think I should spend some time to learn Julia.
